vmdb/app/models/host.rb

```diff
@@ -789,8 +789,9 @@ def refreshable_status
   def scannable_status
     s = self.refreshable_status
     if s[:show] == false && s[:enabled] == false
-      if self.authentication_valid?(:ipmi) || !self.ipmi_address.blank?
-        if self.authentication_invalid?(:ipmi)
+      # TODO: Simplify these conditions
+      if self.has_credentials?(:ipmi) || !self.ipmi_address.blank?
+        if self.missing_credentials?(:ipmi)
```

Comment:
Yeah, I need a truth table.  We need to put this under test and see what bugs shake out.  At the very least, it's hard to comprehend why we do

```if self.authentication_valid?(:ipmi) || !self.ipmi_address.blank?```

Then do
```if self.ipmi_address.blank?```